### DISCUSSION TITLE
@oneOf directive support

### DISCUSSION THREAD

**Discussion by drenizg** [Netflix/dgs-framework#1642] [General]
Hi,

Considering that graphql-java project already supports `@oneOf` directive according to GraphQL draft spec (please see graphql-java/graphql-java#3275), is there any interest/plan to support this on DGS framework?

Alternatively, would it be advisable to implement this as a custom directive, as described here: https://netflix.github.io/dgs/advanced/custom-directives/ ?

Thanks in advance for any guidance you can give!

**Comment by srinivasankavitha** [comment]
Hi @drenizg - you should be able to use all the directives from graphql-java as is when we pick up the latest graphql-java release that has this update.

**Reply by drenizg** [reply]
Oh I see, good to know. Unfortunately, this directive doesn't seem to be released yet. As a temporary solution, would it be possible to implement this behavior as a custom directive?

**Reply by srinivasankavitha** [reply]
Should be possible to add in your application. We wouldn't add it in the DGS framework since it will be available soon anyway.

**Reply by drenizg** [reply]
Right, I was thinking in my application, just wasn't sure if this could be done that way. Thanks a lot!

**Comment by drenizg** [comment]
Hi @srinivasankavitha! I'm reopening just to let you know there's a new [release](https://github.com/graphql-java/graphql-java/releases/tag/v21.2) that adds (experimental) support for this directive.

**Reply by srinivasankavitha** [reply]
Thanks, yes we will be picking up this release in the next few weeks.
